p2panda-rs/src/document/document.rs

```diff
@@ -309,169 +314,171 @@ mod tests {
-                &panda_entry_1_hash.clone().into(),
-            ),
-        )
-        .await
-        .unwrap();
+
+        let panda_operation_2 = OperationBuilder::new(schema.id())
+            .action(OperationAction::Update)
+            .fields(&[("name", OperationValue::String("Panda Cafe!".to_string()))])
+            .previous_operations(&panda_entry_1.hash().into())
+            .build()
+            .unwrap();
+
+        let (panda_entry_2, _) = send_to_store(&mut store, &panda_operation_2, &schema, &panda)
+            .await
+            .unwrap();
 
         // Penguin publishes an update operation which creates a new branch in the graph.
         // This is because they didn't know about Panda's second operation.
         //
         // DOCUMENT: [panda_1]<--[penguin_1]
         //                    \----[panda_2]
-        let (penguin_entry_1_hash, _) = send_to_node(
-            &mut node,
-            &penguin,
-            &update_operation(
-                &[("name", OperationValue::Text("Penguin Cafe!!!".to_string()))],
-                &panda_entry_1_hash.clone().into(),
-            ),
-        )
-        .await
-        .unwrap();
+
+        let penguin_operation_1 = OperationBuilder::new(schema.id())
+            .action(OperationAction::Update)
+            .fields(&[(
+                "name",
+                OperationValue::String("Penguin Cafe!!!".to_string()),
+            )])
+            .previous_operations(&panda_entry_1.hash().into())
+            .build()
+            .unwrap();
+
+        let (penguin_entry_1, _) =
+            send_to_store(&mut store, &penguin_operation_1, &schema, &penguin)
+                .await
+                .unwrap();
 
         // Penguin publishes a new operation while now being aware of the previous branching situation.
         // Their `previous_operations` field now contains 2 operation id's.
         //
         // DOCUMENT: [panda_1]<--[penguin_1]<---[penguin_2]
         //                    \----[panda_2]<--/
-        let (penguin_entry_2_hash, _) = send_to_node(
-            &mut node,
-            &penguin,
-            &update_operation(
-                &[("name", OperationValue::Text("Polar Bear Cafe".to_string()))],
-                &DocumentViewId::new(&[
-                    penguin_entry_1_hash.clone().into(),
-                    panda_entry_2_hash.clone().into(),
-                ])
-                .unwrap(),
-            ),
-        )
-        .await
-        .unwrap();
+
+        let penguin_operation_2 = OperationBuilder::new(schema.id())
+            .action(OperationAction::Update)
+            .fields(&[(
+                "name",
+                OperationValue::String("Polar Bear Cafe".to_string()),
+            )])
+            .previous_operations(&DocumentViewId::new(&[
+                penguin_entry_1.hash().into(),
+                panda_entry_2.hash().into(),
+            ]))
+            .build()
+            .unwrap();
```

Comment:
This looks so nice! We could even make use of the "simple operation" conversion traits for operation values to make it even nicer, but thats not necessary.

Reply:
I'll look into that!